layers/+lang/lean/config.el

```diff
@@ -0,0 +1,16 @@
+;;; config.el --- Lean Layer config File for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(configuration-layer/declare-layers '(auto-completion lua syntax-checking))
+
```

Comment:
This may be not be the ideal way.
Take a look at other layers how they define auto-completion(e.g. python layer), syntax-checking. You should not load layers forcefully when it can be optional. If this layer only supports company completion, then add company package in layer packages and configure it in post-init-company when auto-completion layer is used by the user. Same goes for flycheck(syntax-checking).